proposals/dao/fip_79a.ts

```diff
@@ -0,0 +1,41 @@
+import { ethers } from 'hardhat';
+import chai, { expect } from 'chai';
+import CBN from 'chai-bn';
+import { DeployUpgradeFunc, SetupUpgradeFunc, TeardownUpgradeFunc, ValidateUpgradeFunc } from '../../types/types';
+import { Core, FeiDAO } from '@custom-types/contracts';
+
+chai.use(CBN(ethers.BigNumber));
+
+/*
+DAO ACTIONS:
+1. Grant Governor role to Rari DAO
+2. Change the DAO timelock from the newTimelock to the oldTimelock
+*/
+
+export const deploy: DeployUpgradeFunc = async (deployAddress, addresses, logging = false) => {
+  logging && console.log('No deploy');
+  return {};
+};
+
+export const setup: SetupUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  logging && console.log('No setup');
+};
+
+export const teardown: TeardownUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  logging && console.log('No teardown');
+};
+
+export const validate: ValidateUpgradeFunc = async (addresses, oldContracts, contracts) => {
+  const GOVERN_ROLE = ethers.utils.id('GOVERN_ROLE');
+  const core: Core = contracts.core as Core;
+
+  // 1. Validate that Rari timelock has governor role
+  expect(await core.hasRole(GOVERN_ROLE, addresses.rariTimelock)).to.be.true;
+
+  // 2. Validate that oldTimelock does not have governor role
+  expect(await core.hasRole(GOVERN_ROLE, addresses.timelock)).to.be.false;
+
+  // 3. Validate that Fei DAO timelock has been changed to oldTimelock
+  const feiDAO: FeiDAO = contracts.feiDAO as FeiDAO;
+  expect(await feiDAO.timelock()).to.be.equal(addresses.timelock);
```

Comment:
From my understanding of how our testing flow currently works:

1. Write a set of e2e tests for a new feature. These can take in a `proposals` object to apply upgrades locally and then validate using unit tests that the e2e flow works as expected

2. Seperately, make a `proposals/dao/fip_x.ts` script. Then:
 - Submit the proposal on chain
 - Run `DEPLOY_FILE=fip_x npm run check-proposal`. 
 - This will run the `deploy()` and `setup()` functions of the `proposals/dao/fip_x.ts` script 
 - It will also execute the proposal that was queued on chain, locally
 - After this, the `teardown()` and `validate()` functions will be called to check that the proposal which has been made on mainnet to our DAO will work as expected. So at this stage, the `validate()` tests will pass

Reply:
I adjusted the `TestEndToEndCoordinator` so that the `proposals` constructor argument is optional. 

This allows us to manually simulate and test the proxy migration in `migrateProxies/index.ts`. Then we can also run the `fip_79a.ts` scripts in the other e2e tests + as part of `check-proposal`